### PR TITLE
Fixes #115

### LINKED ISSUE #115: No auto delete when fd is closed on Windows
On Windows (10), the auto-deletion of temporary files doesn't work if the file descriptor is closed manually.

The reason for this is that the "calculation" of `EBADF` doesn't produce the correct result on Windows. It's set to ~~`-4083`~~ `9` while it should be ~~`9`~~ `4083`. ~~The property `os.constants.errno.EBADF` contains the correct value.~~

I'm fairly new to NodeJS ~~so I don't feel like I could provide a proper fix (pull request) - especially considering backward compatibility with older node versions.~~ and don't know how this should/could be solved properly.

Here's some code to reproduce the problem:

```javascript
const fs = require('fs');
const os = require('os');
const _c = process.binding('constants');

const EBADF = _c.EBADF || _c.os.errno.EBADF; // Like the tmp module does it.

const CREATE_FLAGS = (_c.O_CREAT || _c.fs.O_CREAT) | (_c.O_EXCL || _c.fs.O_EXCL) | (_c.O_RDWR || _c.fs.O_RDWR);
const FILE_MODE = 384;

fs.unlinkSync('test.txt');

var fd = fs.openSync('test.txt', CREATE_FLAGS, FILE_MODE);
fs.writeSync(fd, 'test');
fs.closeSync(fd);

try {
    fs.closeSync(fd);
}
catch (e) {
    console.log('Exception: ' + e);
    console.log('ErrNo: ' + e.errno);
    console.log('Expected: ' + EBADF);
    console.log('From Constants: ' + os.constants.errno.EBADF);
}
```

This produces the following output:

```
Exception: Error: EBADF: bad file descriptor, close
ErrNo: -4083
Expected: 9
From Constants: 9
```

### LINKED ISSUE #117: detachDescriptor causes error with fixes for #115 in place
The problem here is that in the removeCallback it tests for fdPath[0] being LTE 0, which is wrong, it should read -1 != fdPath[0].

This might have been found due to the side effect caused by moving the unlinkSync() back into the try block, which had been moved out of that block, causing the fix for #115 to fail.

### PR DESCRIPTION
**Pull request by silkentrance**
Fixes #117
Add node 7 to travis build matrix
Bump version to 0.0.32

### CODE REVIEW

File: lib/tmp.js [hunk 1]
```diff
@@ -367,20 +369,28 @@ function dirSync(options) {
 function _prepareTmpFileRemoveCallback(name, fd, opts) {
   const removeCallback = _prepareRemoveCallback(function _removeCallback(fdPath) {
     try {
-      if (0 <= fdPath[0]) {
+      if (-1 != fdPath[0]) {
```
Comment (raszi) [inline lib/tmp.js]: Is there any specific reason why you allow negative values as well except `-1`?

Comment (silkentrance) [inline lib/tmp.js]: There was this other PR that sets the fd to -1.

Comment (raszi) [inline lib/tmp.js]: Right now the condition in the master branch looks like this:

```javascript
if (0 <= fdPath[0])
```

Comment (silkentrance) [inline lib/tmp.js]: Ah, I now see the problem. I will change that as soon as I find the time.

**Comment by silkentrance** [comment]
@raszi What do you think? All tests have passed. Shall we merge and close both #117 and #115?

**Comment by raszi** [comment]
Sure, yeah! I just fixed the small file descriptor issue but I did not have time to wait for the tests to finish on Travis.

**Pull request merged**